src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc

```diff
@@ -154,6 +211,19 @@ class XdsResolver : public Resolver {
     if (xds_client_ != nullptr) xds_client_->ResetBackoff();
   }
 
+  void MaybeRemoveUnusedClusters();
+
+  void Run(std::function<void()> callback, DebugLocation location) {
+    work_serializer_->Run(std::move(callback), location);
+  }
+
+  ClusterState::ClusterStateMap::iterator Emplace(
```

Comment:
I don't think we should need this method.  It looks like there's some legacy complexity here related to how we're storing the cluster name in the map keys.  I've sent you #32989 to simplify this.  Once that change has been merged, I don't think we'll need this method anymore.